repository/src/main/java/org/apache/atlas/repository/store/graph/v2/preprocessor/datamesh/DataProductPreProcessor.java

```diff
@@ -186,7 +218,127 @@ private static String createQualifiedName(String parentDomainQualifiedName) thro
         if (StringUtils.isEmpty(parentDomainQualifiedName)) {
             throw new AtlasBaseException(AtlasErrorCode.BAD_REQUEST, "Parent Domain Qualified Name cannot be empty or null");
         }
-        return parentDomainQualifiedName + "/product/" + getUUID();
+        return parentDomainQualifiedName + "/product/" + PreProcessorUtils.getUUID();
+
+    }
+
+    private void createDaapVisibilityPolicy(AtlasEntity entity,AtlasVertex vertex) throws AtlasBaseException {
+        String productGuid = vertex.getProperty("__guid", String.class);
+
+        AtlasEntity policy = new AtlasEntity();
+        policy.setTypeName(POLICY_ENTITY_TYPE);
+        policy.setAttribute(NAME, entity.getAttribute(NAME));
+        policy.setAttribute(QUALIFIED_NAME, productGuid + "/read-policy");
+        policy.setAttribute(ATTR_POLICY_ACTIONS, Arrays.asList("entity-read"));
+        policy.setAttribute(ATTR_POLICY_CATEGORY, MESH_POLICY_CATEGORY);
+        policy.setAttribute(ATTR_POLICY_TYPE, POLICY_TYPE_ALLOW);
+        policy.setAttribute(ATTR_POLICY_RESOURCES, Arrays.asList("entity:" + entity.getAttribute(QUALIFIED_NAME)));
+        policy.setAttribute(ATTR_POLICY_RESOURCES_CATEGORY, POLICY_RESOURCE_CATEGORY_PERSONA_ENTITY);
+        policy.setAttribute(ATTR_POLICY_SERVICE_NAME, "atlas");
+        policy.setAttribute(ATTR_POLICY_SUB_CATEGORY, DATA_PRODUCT); // create new constant attr
```

Comment:
Let us move this to another helper method

Reply:
Also, overall add a if check here to only do these things if visibility is not PRIVATE